jax/_src/scipy/special.py

```diff
@@ -1563,84 +1563,71 @@ def _gen_associated_legendre(l_max: int,
   Args:
     l_max: The maximum degree of the associated Legendre function. Both the
       degrees and orders are `[0, 1, 2, ..., l_max]`.
-    x: A vector of type `float32`, `float64` containing the sampled points in
-      spherical coordinates, at which the ALFs are computed; `x` is essentially
-      `cos(θ)`. For the numerical integration used by the spherical harmonics
-      transforms, `x` contains the quadrature points in the interval of
-      `[-1, 1]`. There are several approaches to provide the quadrature points:
-      Gauss-Legendre method (`scipy.special.roots_legendre`), Gauss-Chebyshev
-      method (`scipy.special.roots_chebyu`), and Driscoll & Healy
-      method (Driscoll, James R., and Dennis M. Healy. "Computing Fourier
-      transforms and convolutions on the 2-sphere." Advances in applied
-      mathematics 15, no. 2 (1994): 202-250.). The Gauss-Legendre quadrature
-      points are nearly equal-spaced along θ and provide exact discrete
-      orthogonality, (P^m)^T W P_m = I, where `T` represents the transpose
-      operation, `W` is a diagonal matrix containing the quadrature weights,
-      and `I` is the identity matrix. The Gauss-Chebyshev points are equally
-      spaced, which only provide approximate discrete orthogonality. The
-      Driscoll & Healy quadrature points are equally spaced and provide the
-      exact discrete orthogonality. The number of sampling points is required to
-      be twice as the number of frequency points (modes) in the Driscoll & Healy
-      approach, which enables FFT and achieves a fast spherical harmonics
-      transform.
-    is_normalized: True if the associated Legendre functions are normalized.
-      With normalization, `N_l^m` is applied such that the spherical harmonics
-      form a set of orthonormal basis functions of L^2(S^2).
+    x: A vector containing the evaluation points in `[-1, 1]`.
+    is_normalized: Whether to apply spherical-harmonic normalization.
 
   Returns:
-    The 3D array of shape `(l_max + 1, l_max + 1, len(x))` containing the values
-    of the ALFs at `x`; the dimensions in the sequence of order, degree, and
-    evaluation points.
+    A 3D array of shape `(l_max + 1, l_max + 1, len(x))` containing the ALFs.
   """
+
   p = jnp.zeros((l_max + 1, l_max + 1, x.shape[0]), dtype=x.dtype)
 
-  a_idx = jnp.arange(1, l_max + 1, dtype=x.dtype)
-  b_idx = jnp.arange(l_max, dtype=x.dtype)
+  sqrt1mx = jnp.sqrt(jnp.clip(1.0 - x * x, 0.0))
+
+  # Always start unnormalized
+  p00 = 1.0
+  p = p.at[(0, 0)].set(p00)
```

Comment:
Stylistically, we typically do not explicitly pass tuples to indexing syntax:
```python
  p = p.at[0, 0].set(p00)
```
Same in a number of places below.